doc/specs/#888 - Com Api.md

```diff
@@ -0,0 +1,812 @@
+
+# 1. Background
+
+The Windows Package Manager currently exposes a command line interface to search for packages, install them, view progress, and more. 
+This API is designed to provide another way for callers to make use of that functionality. The API will be preferred by 
+callers that want to receive progress and completion events, and UWP packages that do not have permission to launch command line 
+processes. The goal for this api is to provide the full set of install functionality possible using the Windows Package Manager command line.
+The command line is documented at https://docs.microsoft.com/en-us/windows/package-manager/winget/
+
+# 2. Description
+
+Windows Package Manager is a package manager for windows applications. It comes with a predefined repository of applications and users can add 
+new repositories using the winget command line. This API allows packaged apps with the packageManagement capability and other higher 
+privilege processes to start, manage, and monitor installation of packages that are listed in Windows Package Manager repositories.
+
+# 3. Examples
+
+Sample member values for the following examples:
+
+```c++ (C++ish pseudocode)
+m_installAppId = L"Microsoft.VSCode";
+```
+
+## 3.1. Create objects
+
+Creation of objects has to be done through CoCreateInstance rather than normal winrt initialization since it's hosted by an out of proc com server.
+These helper methods will be used in the rest of the examples.
+
+```c++ (C++ish pseudocode)
+    AppInstaller CreateAppInstaller()
+    {
+        com_ptr<abi::IAppInstaller> abiAppInstaller = nullptr;
+        check_hresult(CoCreateInstance(CLSID_AppInstaller, nullptr, CLSCTX_LOCAL_SERVER, IID_PPV_ARGS(abiAppInstaller.put())));
+        check_pointer(abiAppInstaller.get());

```

Comment:
Remove this check for null; CoCreateInstance should return an error if the instance can't be cocreated.